Rust fails to build on Windows.

# What's the problem?

When I try to compile my program with libmem, cargo tells me that the compilation failed:

```
Creating library C:\<my_program_dir>\target\release\deps\cetutorial.lib and object C:\<my_program_dir>\target\release\deps\cetutorial.exp
          LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
          libmem.lib(Path.cpp.obj) : error LNK2019: unresolved external symbol SHGetKnownFolderPath referenced in function "bool __cdecl llvm_ks::sys::path::getUserCacheDir(class llvm_ks::SmallVectorImpl<char> &)" (?getUserCacheDir@path@sys@llvm_ks@@YA_NAEAV?$SmallVectorImpl@D@3@@Z)
          C:\<my_program_dir>\target\release\deps\cetutorial.exe : fatal error LNK1120: 1 unresolved externals
```

Do you have a good solution for this please?

# My information

My OS version: Windows 11 Home Edition 23H2

My computer: HP ENVY Laptop

My Rust version:rustc 1.79.0 (129f3b996 2024-06-10)

My Rustup version: rustup 1.27.1 (54dd3d00f 2024-04-24)

libmem version: libmem = "5.0.0"

using MSVC toolchain

--- a/bindings/rust/libmem-sys/build.rs
+++ b/bindings/rust/libmem-sys/build.rs
@@ -105,7 +105,7 @@ fn main() {
     // Resolve link dependencies
     let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
     let deps = if target_os == "windows" {
-        vec!["user32", "psapi", "ntdll"]
+        vec!["user32", "psapi", "ntdll", "shell32"]
     } else if target_os == "linux" {
         vec!["dl", "m", "stdc++"]
     } else if target_os == "freebsd" {

